File: newsroom-flow-android/app/src/main/java/com/newsroomflow/android/data/ServiceLocator.kt
package com.newsroomflow.android.data

import android.content.Context

/**
 * Tiny manual DI container — initialised once from NewsroomFlowApplication.
 */
object ServiceLocator {
    lateinit var authStore: AuthStore
        private set
    lateinit var apiClient: ApiClient
        private set

    fun init(context: Context) {
        authStore = AuthStore(context.applicationContext)
        apiClient = ApiClient(tokenProvider = { authStore.token })
    }

    val api: NewsroomFlowApi get() = apiClient.api
}

File: newsroom-flow-ios/NewsroomFlow/Data/ServiceLocator.swift
import Foundation

/// Tiny manual DI container — the iOS counterpart of the Android
/// `ServiceLocator` object. Constructed once at app launch and shared
/// app-wide. The API client reads the bearer token lazily from the same
/// AuthStore instance, so a fresh login is picked up on the next request.
final class ServiceLocator {
    static let shared = ServiceLocator()

    let authStore: AuthStore
    let api: NewsroomFlowAPI

    private init() {
        let store = AuthStore()
        self.authStore = store
        self.api = NewsroomFlowAPI(tokenProvider: { [weak store] in store?.token })
    }
}
